### DISCUSSION TITLE
DataTable: Detecting Page Change

### DISCUSSION THREAD

**Discussion by SolidAnonDev** [primefaces/community#1613] [PrimeReact] [answered]
I have `dragSelection` and `selectionMode="multiple"` enabled on my datatable.

I want to reset the selection when the user changes the page, so there can't be multiple selections across multiple pages. The easiest way to get this done it would seem is to listen for a page change in a `useEffect` or something but I can't seem to figure out how to access that.

I had hoped the `onPage` prop would work, but that appears to **cancel the native paging functionality**, meaning I can't change the page at all if I've provided that prop and I don't intend to provide logic to handle paging, I want to use the inbuilt paging.

Does anyone have any idea how I could access something to determine when the page has changed? It would seem like it might exist in the same place as changing the page programtically, but I can't find any info on that either..

**Comment by melloware** [comment, marked as answer]
Yes you should be able to listen to onValueChange: https://primereact.org/datatable/#api.DataTable.callbacks.onValueChange

That gets call on sorting, filtering, or paging.

**Reply by SolidAnonDev** [reply]
Hey thanks, this will actually solve two problems, because my next thought was how to do it when the filters change. Thanks a bunch!
